Makefile

```diff
@@ -36,10 +36,16 @@ endif
 
 PKG=sigs.k8s.io/release-utils/version
 LDFLAGS='"-X $(PKG).gitVersion=$(GIT_VERSION) -X $(PKG).gitCommit=$(GIT_HASH) -X $(PKG).gitTreeState=$(GIT_TREESTATE) -X $(PKG).buildDate=$(BUILD_DATE)"'
+export CGO_ENABLED=1
 
 .PHONY: kpromo
 kpromo:
-	${REPO_ROOT}/go_with_version.sh build -trimpath -ldflags '-s -w -buildid= -extldflags "-static" $(LDFLAGS)' -o ./bin/kpromo ./cmd/kpromo
+	go build \
+		-trimpath \
+		-ldflags '-s -w -buildid= -linkmode=external -extldflags=-static $(LDFLAGS)' \
```

Comment:
Can we document (e.g. in the PR description) why do we need static linking?

Reply:
It was using static linking before via `-extldflags "-static"`, but this had no effect when running locally :shrug: 

The image we use is already relying on static linking: 
```
> podman run -it --entrypoint sh registry.k8s.io/artifact-promoter/kpromo:v3.6.0-0 -c 'ldd /kpromo'
        not a dynamic executable
```

Reply:
Thanks for clarification!